crates/aptos/src/account/key_rotation.rs

```diff
@@ -0,0 +1,259 @@
+
+        eprintln!("{}", string);
+
+        if let Some(txn_success) = txn_summary.success {
+            if !txn_success {
+                return Err(CliError::ApiError(
+                    "transaction was not executed successfully".to_string(),
+                ));
+            }
+        } else {
+            return Err(CliError::UnexpectedError(
+                "Mailformed transaction response".to_string(),
+            ));
+        }
+
+        // Asks user if they want to create a new Profile. Overriding profile is a bit of risky, we create a new one
+        // instead.
+
+        if let Err(cli_err) = prompt_yes_with_override(
+            "Do you want to create a profile with the new private key?",
+            self.prompt_options,
+        ) {
+            match cli_err {
+                CliError::AbortedError => {
+                    return Ok(());
+                }
+                _ => {
+                    return Err(cli_err);
+                }
+            }
+        }
+
+        eprintln!("Enter the name for the profile");
+        let profile_name = read_line("Profile name")?.trim().to_string();
+
+        if profile_name.is_empty() {
+            return Ok(());
+        }
+
+        let profile_config = ProfileConfig {
+            private_key: Some(new_private_key.clone()),
+            public_key: Some(new_private_key.public_key()),
+            ..self.txn_options.profile_options.profile()?
+        };
+
+        let mut config = CliConfig::load(ConfigSearchMode::CurrentDirAndParents)?;
+
+        if config.profiles.is_none() {
+            // This should not happen. The command requires a profile exist to rotate key
+            config.profiles = Some(BTreeMap::new());
+        }
+        config
+            .profiles
+            .as_mut()
+            .unwrap()
+            .insert(profile_name.clone(), profile_config);
+        config.save()?;
+
+        eprintln!("Profile {} is saved.", profile_name);
+
+        Ok(())
+    }
+}
+
+/// Command to lookup the account adress through on-chain lookup table
+///
+#[derive(Debug, Parser)]
+pub struct LookupAddress {
+    #[clap(flatten)]
+    pub(crate) encoding_options: EncodingOptions,
+
+    #[clap(flatten)]
+    pub(crate) public_key_options: PublicKeyInputOptions,
+
+    #[clap(flatten)]
+    pub(crate) profile_options: ProfileOptions,
+
+    #[clap(flatten)]
+    pub(crate) rest_options: RestOptions,
+}
+
+impl LookupAddress {
+    pub(crate) fn public_key(&self) -> CliTypedResult<Ed25519PublicKey> {
+        self.public_key_options.extract_public_key(
+            self.encoding_options.encoding,
+            &self.profile_options.profile,
+        )
+    }
+
+    /// Builds a rest client
+    fn rest_client(&self) -> CliTypedResult<Client> {
+        self.rest_options.client(&self.profile_options.profile)
+    }
+}
+
+#[async_trait]
+impl CliCommand<String> for LookupAddress {
+    fn command_name(&self) -> &'static str {
+        "LookupAddress"
+    }
```

Comment:
If this command gets the account on chain, should we be using a version of this for all other commands?  Or do we expect the address in the profile to always be right?

Reply:
Account addresses do not change in profiles. This command is meant for people to lookup the address if they only have the keys.